File: base-class-module/src/main/java/jingya/com/base_class_module/BaseAdapters/BaseFragmentPagerAdapter.java
package jingya.com.base_class_module.BaseAdapters;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.List;

/**
 * @author kuky
 * @description viewpager adapter 封装
 */
public class BaseFragmentPagerAdapter extends FragmentPagerAdapter {

    private List<Fragment> mList;
    private String[] mTitles;

    public BaseFragmentPagerAdapter(FragmentManager fm, List<Fragment> list, String[] titles) {
        super(fm);
        this.mList = list;
        this.mTitles = titles;
    }

    @Override
    public Fragment getItem(int position) {
        return mList.get(position);
    }

    @Override
    public int getCount() {
        return mList.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mTitles == null ? super.getPageTitle(position) : mTitles[position];
    }
}

File: base-kt-module/src/main/java/com/kuky/base_kt_module/BaseAdapters/BaseFragmentPagerAdapter.kt
package com.kuky.base_kt_module.BaseAdapters

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

/**
 * @author Kuky
 */
class BaseFragmentPagerAdapter(fm: FragmentManager, fragments: MutableList<Fragment>, titles: Array<String>)
    : FragmentPagerAdapter(fm) {

    private var mFragments: MutableList<Fragment>? = null
    private var mTitles: Array<String>? = null

    init {
        this.mFragments = fragments
        this.mTitles = titles
    }

    override fun getItem(position: Int): Fragment {
        return mFragments!![position]
    }

    override fun getCount(): Int {
        return mTitles!!.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return if (mTitles!!.isEmpty()) super.getPageTitle(position)
        else mTitles!![position]
    }
}